app/evmante/evmante_can_transfer.go

```diff
@@ -59,37 +57,27 @@ func (ctd CanTransferDecorator) AnteHandle(
 			return ctx, errors.Wrapf(
 				sdkerrors.ErrInsufficientFee,
 				"gas fee cap (wei) less than block base fee (wei); (%s < %s)",
-				coreMsg.GasFeeCap(), baseFeeWeiPerGas,
+				evmMsg.GasFeeCap(), baseFeeWeiPerGas,
 			)
 		}
 
-		cfg := &statedb.EVMConfig{
-			ChainConfig: ethCfg,
-			Params:      params,
-			// Note that we use an empty coinbase here  because the field is not
-			// used during this Ante Handler.
-			BlockCoinbase: gethcommon.Address{},
-			BaseFeeWei:    baseFeeWeiPerGas,
-		}
-
-		stateDB := ctd.NewStateDB(
```

Comment:
Creating a new StateDB here uses the anteCtx, which gets tossed when the tx runs. That means whenever StateDB commits, it's committing to a out of scope ctx.

It's better to just read the bank balances directly via the Cosmos ctx (which is what core.CanTransfer does anyways).